src/libs/auth.ts

```diff
@@ -97,49 +100,68 @@ const revokeTokens = async () => {
 const getSigninArgs = (includeBillingScope) => {
   return Utils.cond(
     [includeBillingScope === false, () => ({})],
-    // For Google just append the scope to the signin args.
-    [isGoogleAuthority(), () => ({ scope: 'openid email profile https://www.googleapis.com/auth/cloud-billing' })],
-    // For B2C switch to a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    // For B2C use a dedicated policy endpoint configured for the GCP cloud-billing scope.
     () => ({
       extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
       extraTokenParams: { p: getConfig().b2cBillingPolicy },
     })
   );
 };
 
-export const signIn = async (includeBillingScope = false) => {
+export const signIn = async (includeBillingScope = false): Promise<User> => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
-  // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
-  // is no way to determine it after the fact.
-  // For Google we don't need to do this since we can inspect the scope directly in the user object.
-  if (!isGoogleAuthority()) {
-    authStore.update((state) => ({ ...state, hasGcpBillingScopeThroughB2C: includeBillingScope }));
-  }
-  Ajax().Metrics.captureEvent(Events.userLogin, { authProvider: user.profile.idp });
+  const user: User = await getAuthInstance().signinPopup(args);
+
+  const sessionId = uuid();
+  const sessionStartTime: number = Date.now();
+  const userJWT: any = user.id_token;
```

Comment:
```suggestion
  const userJWT: string = user.id_token!;
```

`user.id_token` is typed `string | undefined`.
https://authts.github.io/oidc-client-ts/classes/User.html#id_token

We know it's there because we specified the "openid" scope, so we can assert it's non-null.